Signal/src/AppDelegate.m

```diff
@@ -52,6 +53,23 @@ - (BOOL)application:(UIApplication *)application didFinishLaunchingWithOptions:(
     CategorizingLogger *logger = [CategorizingLogger categorizingLogger];
     [logger addLoggingCallback:^(NSString *category, id details, NSUInteger index){
     }];
+    
+    BOOL loggingIsEnabled;
+    
+#ifdef DEBUG
+    // Specified at Product -> Scheme -> Edit Scheme -> Test -> Arguments -> Environment to avoid things like
+    // the phone directory being looked up during tests.
+    loggingIsEnabled = TRUE;
+    [DebugLogger.sharedLogger enableTTYLogging];
+#elif RELEASE
+    loggingIsEnabled = Environment.preferences.loggingIsEnabled;
```

Comment:
This looks a little funny since we're now accessing something on the environment before we've "set" the environment. It might work since it's ultimately the storage singleton, but at a minimum, can you confirm you've tested the behavior in a release build with file logging enabled/disabled?

Toggle file logging in (settings > privacy > enable debug log), and read the contents of the debug log by 
1. emailing it to yourself 
2. pretending like you're going to submit a GH issue which just copies it to your pasteboard buffer.